velox/exec/Window.cpp

```diff
@@ -573,9 +615,26 @@ vector_size_t Window::callApplyLoop(
 
   // This function requires that the currentPartition_ is available for output.
   VELOX_DCHECK_NOT_NULL(currentPartition_);
+
+  // Used to record the processed row count in current partition. And then
+  // passed this value to clearOutputRows to clear the real row count.
+  auto processedRowCount = -1;
   while (numOutputRowsLeft > 0) {
-    auto rowsForCurrentPartition =
-        currentPartition_->numRows() - partitionOffset_;
+    auto totalRows = currentPartition_->numRows();
+    auto rowsForCurrentPartition = totalRows - partitionOffset_;
+
+    // For a partial window partition, currentPartition_->numRows() returns the
+    // number of remaining rows in the current partition that have not been
+    // processed.
+    if (currentPartition_->isPartial()) {
+      rowsForCurrentPartition = totalRows;
+
+      // Delete the last row number in previous batch.
+      if (partitionOffset_ > 0) {
```

Comment:
This is confusing. It might be better to add a new method in WindowPartition to return the numRowsForProcessing as numRows (for first output) or numRows - 1 for next output and use that in the code instead of these adjustments.

Reply:
@aditi-pandit Updated.